src/main/java/com/synopsys/integration/alert/channel/jira/JiraIssueHandler.java

```diff
@@ -0,0 +1,253 @@
+import com.synopsys.integration.jira.common.cloud.model.response.IssueResponseModel;
+import com.synopsys.integration.jira.common.cloud.model.response.IssueSearchResponseModel;
+import com.synopsys.integration.jira.common.cloud.model.response.IssueTypeResponseModel;
+import com.synopsys.integration.jira.common.cloud.model.response.PageOfProjectsResponseModel;
+import com.synopsys.integration.jira.common.cloud.model.response.TransitionsResponseModel;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssuePropertyService;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssueSearchService;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssueService;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssueTypeService;
+import com.synopsys.integration.jira.common.cloud.rest.service.ProjectService;
+
+public class JiraIssueHandler {
+    private static final Logger logger = LoggerFactory.getLogger(JiraIssueHandler.class);
+
+    private ProjectService projectService;
+    private IssueService issueService;
+    private IssueSearchService issueSearchService;
+    private IssueTypeService issueTypeService;
+    private JiraProperties jiraProperties;
+
+    private JiraIssuePropertyHelper jiraIssuePropertyHelper;
+
+    public JiraIssueHandler(ProjectService projectService, IssueService issueService, IssueSearchService issueSearchService, IssuePropertyService issuePropertyService, IssueTypeService issueTypeService, JiraProperties jiraProperties) {
+        this.projectService = projectService;
+        this.issueService = issueService;
+        this.issueSearchService = issueSearchService;
+        this.issueTypeService = issueTypeService;
+        this.jiraProperties = jiraProperties;
+        this.jiraIssuePropertyHelper = new JiraIssuePropertyHelper(issueSearchService, issuePropertyService);
+    }
+
+    public String createOrUpdateIssues(FieldAccessor fieldAccessor, MessageContentGroup content) throws IntegrationException {
+        final String projectName = fieldAccessor.getString(JiraDescriptor.KEY_JIRA_PROJECT_NAME).orElse("");
+        final PageOfProjectsResponseModel projectsResponseModel = projectService.getProjectsByName(projectName);
+        final ProjectComponent projectComponent = projectsResponseModel.getProjects()
+                                                      .stream()
+                                                      .findFirst()
+                                                      .orElseThrow(() -> new AlertException("Expected to be passed an existing project name."));
+
+        final String projectId = projectComponent.getId();
+        final String projectKey = projectComponent.getKey();
+        final String issueType = fieldAccessor.getString(JiraDescriptor.KEY_ISSUE_TYPE).orElse(JiraDistributionUIConfig.DEFAULT_ISSUE_TYPE);
+        final boolean matchingIssueTypeFound = issueTypeService.getAllIssueTypes()
```

Comment:
It looks like 'matchingIssueTypeFound' is never used. If we can't find the Issue Type shouldn't we throw an exception the way we do for the Jira Project and the Provider?

Reply:
Yes, that was unused before this PR. The purpose of this PR is to add appropriate abstraction to what existed and begin working on the issue remapping functionality. I'll go ahead and take care of this anyway, but that's why it hadn't been touched.